internal/gcsx/file_cache_reader_test.go

```diff
@@ -85,7 +87,13 @@ func (t *fileCacheReaderTest) TearDown() {
 func (t *fileCacheReaderTest) mockNewReaderWithHandleCallForTestBucket(start uint64, limit uint64, rd gcs.StorageReader) {
 	t.mockBucket.On("NewReaderWithReadHandle", mock.Anything, mock.MatchedBy(func(rg *gcs.ReadObjectRequest) bool {
 		return rg != nil && (*rg.Range).Start == start && (*rg.Range).Limit == limit
-	})).Return(rd, nil).Maybe()
+	})).Return(rd, nil).Once()
```

Comment:
For reviewer: It was causing issue as again calling the method was reusing the old reader.